File: BEST-App-main/lib/model/product_detail.dart
class ProductDetail {
   int id;
   int productSizeId;
   int productColorId;
  int quantity;

  ProductDetail(this.id, this.productSizeId, this.productColorId, this.quantity);

  static Map<String, dynamic> toJson(ProductDetail productDetail) {
    return {
      'id': productDetail.id,
      'product_size_id': productDetail.productSizeId,
      'product_color_id': productDetail.productColorId,
      'quantity': productDetail.quantity,
    };
  }
}

File: spring-best/src/main/java/com/best/springbest/model/ProductDetail.java
package com.best.springbest.model;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "product_detail")
public class ProductDetail {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "product_size_id",nullable = false)
    private ProductSize productSize;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "product_color_id",nullable = false)
    private ProductColor productColor;

    private int quantity;

}
